### DISCUSSION TITLE
Add `+=` operator to pest grammar

### DISCUSSION THREAD

**Discussion by mxmsll** [pest-parser/pest#836] [Ideas]
I am implementing a series of parsers that are based on some common syntax, but all have different extensions. I hope to extend a defined rule by introducing the '+=' operator.
# Example
base.pest
```
base_type_spec = { integer_type }
```
string,pest
```
base_type_spec += { string_type }
```
any.pest
```
base_type_spec += { any_type }
```

parser.rs
```
pub mod parser_with_string {
#[derive(Parser)]
#[grammar = "base.pest"]
#[grammar = "string.pest"]
pub struct MyParser;
}

pub mod parser_with_any {
#[derive(Parser)]
#[grammar = "base.pest"]
#[grammar = "any.pest"]
pub struct MyParser;
}
```

**Comment by tomtau** [comment]
what should that operator do? override `base_type_spec` rule or do `{ integer_type ~ string_type }` ...?

**Reply by mxmsll** [reply]
override `base_type_spec` rule as: { integer_type | string_type }

**Comment by NoahTheDuke** [comment]
I feel this is best handled by being explicit. Inheritance gets complicated quickly. Better to write both versions than try to parse/remember where exactly a given rule is defined.
